src/hotspot/share/opto/macro.cpp

```diff
@@ -347,9 +347,9 @@ Node* PhaseMacroExpand::make_arraycopy_load(ArrayCopyNode* ac, intptr_t offset,
 // on the input paths.
 // Note: this function is recursive, its depth is limited by the "level" argument
 // Returns the computed Phi, or NULL if it cannot compute it.
-Node *PhaseMacroExpand::value_from_mem_phi(Node *mem, BasicType ft, const Type *phi_type, const TypeOopPtr *adr_t, AllocateNode *alloc, Node_Stack *value_phis, int level) {
+Node *PhaseMacroExpand::value_from_mem_phi(Compile* comp, PhaseIterGVN* igvn, Node *mem, BasicType ft, const Type *phi_type, const TypeOopPtr *adr_t, AllocateNode *alloc, Node_Stack *value_phis, int level) {
```

Comment:
What's the reasoning behind passing in `comp `instead of using the instance variable `C`?

Reply:
The `value_from_mem_phi` method was made static so that it can be used without an instance of PhaseMacroExpand. Since the method is static we can't use an instance variable inside it.

Reply:
Ah yes, I forgot this was the one you made static.